File: PigSim/pig_functions.py

from random import randrange

def takeTurn(holdAmount):
    turnScore = 0

    while turnScore < holdAmount:
        rollScore = randrange(1, 7)
        
        if rollScore > 1:
            turnScore += rollScore
        else:
            turnScore = 0
            break
    return turnScore

def takeTurnInteractive():
    turnScore = 0

    while True:
        rollScore = randrange(1, 7)

        if rollScore > 1:
            turnScore += rollScore

            print("You rolled {}".format(rollScore))
            print("Your score for this turn is {}".format(turnScore))

            inputLine = input("Roll again? (Y/N): ")
            inputCaps = inputLine.upper()
            firstChar = inputLine[0]

            if firstChar == "N": break

        else:
            turnScore = 0
            print("You rolled a PIG!")
            print("Your score for the turn is 0")
            break;
    return turnScore

File: PigSim/PigFunctions.java
import java.util.Scanner;

public class PigFunctions {

    private static Scanner input = null;

    public static int takeTurn(int holdAmount) {
        int turnScore = 0;

        while (turnScore < holdAmount) {
            int rollScore = (int) (Math.random() * 6) + 1;

            if (rollScore > 1) {
                turnScore += rollScore;
            } else {
                turnScore = 0;
                break;
            }
        }
        return turnScore;
    }

    public static int takeTurnInteractive() {
        int turnScore = 0;

        //Scanner input = new Scanner(System.in);

        if (input == null) {
            input = new Scanner(System.in);
        }

        while (true) {
            int rollScore = (int) (Math.random() * 6) + 1;

            if (rollScore > 1) {
                turnScore += rollScore;
                System.out.printf("You rolled %d\n", rollScore);
                // format for a decimal number where the spacing doesnt matter
                System.out.printf("Your score for this turn is %d\n", turnScore);

                System.out.print("Roll again? (Y/N): ");
                String inputLine = input.nextLine();
                String inputCaps = inputLine.toUpperCase();
                char firstChar = inputCaps.charAt(0);

                if (firstChar == 'N') {
                    break;
                }
            } else {
                turnScore = 0;
                System.out.println("You rolled a PIG!");
                System.out.println("Your score for the turn is 0");
                break;
            }
        }
        return turnScore;
    }
}